src/main/java/com/google/devtools/build/lib/bazel/rules/cpp/BazelCcLibraryRule.java

```diff
@@ -33,6 +35,17 @@ public RuleClass build(RuleClass.Builder builder, RuleDefinitionEnvironment env)
         // deps, data, linkopts, defines, srcs; override here too?
 
         .requiresConfigurationFragments(CppConfiguration.class)
+        /*<!-- #BLAZE_RULE($cc_rule).ATTRIBUTE(compile_only_deps) -->
+        The list of other libraries to be linked in to the binary target.
+        This allows control over the requirement set as to whether a dependency
+        should contribute to the interface or only compilations of a library
+        <p>These can be <code>cc_library</code>, <code>cc_inc_library</code>, or
+        <code>objc_library</code> targets.</p>
+        <!-- #END_BLAZE_RULE.ATTRIBUTE -->*/
+        .add(attr("compile_only_deps", LABEL_LIST)
+            .allowedRuleClasses(BazelCppRuleClasses.DEPS_ALLOWED_RULES)
```

Comment:
Please change this to allow any rule with the provider CcInfo. Do not allow linker scripts.